v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/schema/IdentityMapper.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.spanner.migrations.schema;
+
+import com.google.cloud.teleport.v2.spanner.ddl.Column;
+import com.google.cloud.teleport.v2.spanner.ddl.Ddl;
+import com.google.cloud.teleport.v2.spanner.ddl.Table;
+import com.google.cloud.teleport.v2.spanner.type.Type;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+/**
+ * This mapper directly maps table and column names without any modification. For fetching spanner
+ * metadata, it uses {@link Ddl}.
+ */
+public class IdentityMapper implements ISchemaMapper {
+
+  Ddl ddl;
+
+  public IdentityMapper(Ddl ddl) {
+    this.ddl = ddl;
+  }
+
+  @Override
+  public String getSpannerTableName(String srcTable) {
+    return srcTable;
+  }
+
+  @Override
+  public String getSpannerColumnName(String srcTable, String srcColumn) {
+    return srcColumn;
+  }
```

Comment:
I didn't get this 🤔

IdentityMapper is supposed to perform mapping via "names".  Therefore - 

1. If a spTable exists with the same name as srcTable, spTable == srcTable should be returned.
2. If an spTable with the same name as srcTable does not exist, then NoSuchElement or null should be returned?

In your implementation, `srcTable` is always returned? 

This is clear from the test case below. 

You passed `abc` value in `srcTable` where the `Ddl` only had a table named `Users`. So the `getSpannerTableName` function should return the corresponding Spanner table name, which is nothing, so NoSuchElementFoundException? 

Instead you return `abc`, which will fail later because there is actually no table named `abc` in Spanner!

Reply:
Yes this would fail at a mutation stage like it does in the live migration template. We could fail faster by adding the check here.

Reply:
Done